gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/GradleJdkInstallationSetup.java

```diff
@@ -111,12 +109,14 @@ private static void setupJdk(StdLogger logger, CaResources caResources, String[]
         }
         Path destinationJdkInstallationDir = Path.of(args[1]);
         Path certsDir = Path.of(args[2]);
-        copy(logger, destinationJdkInstallationDir);
-        Map<String, String> certSerialNumbersToNames = extractCertsSerialNumbers(logger, certsDir);
-        caResources.maybeImportCertsInJdk(destinationJdkInstallationDir, certSerialNumbersToNames);
+        boolean wasCopied = copy(logger, destinationJdkInstallationDir);
+        if (wasCopied) {
+            Map<String, String> certSerialNumbersToNames = extractCertsSerialNumbers(logger, certsDir);
```

Comment:
if we didn't copy the jdk to the destination directory (https://github.com/palantir/gradle-jdks/pull/382/files#diff-d2a5b341108734345bf72bfb7dcb34cdc12976f8ffcf2b99df905702ce8faa12R130), which means that we waited for the lock while another process created the directory, then we shouldn't try to add the certificate because the certificate was already added.

Reply:
I would add a comment about this concurrency stuff.